api/service/cid.go

```diff
@@ -0,0 +1,91 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"gorm.io/gorm"
+
+	"github.com/photon-storage/go-common/log"
+	"github.com/photon-storage/go-photon/crypto/sha256"
+	pbd "github.com/photon-storage/photon-proto/depot"
+
+	"github.com/photo-storage/dropbox/database/orm"
+)
+
+type cidTask struct {
+	ctx      context.Context
+	db       *gorm.DB
+	depotCli pbd.DepotClient
+}
+
+func newCIDTask(
+	ctx context.Context,
+	db *gorm.DB,
+	depotCli pbd.DepotClient,
+) *cidTask {
+	return &cidTask{
+		ctx:      ctx,
+		db:       db,
+		depotCli: depotCli,
+	}
+}
+
+func (c *cidTask) run() {
+	ticker := time.NewTicker(10 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			if err := c.fetchCID(); err != nil {
+				log.Error("fetch object cid failed", "error", err)
+			}
+
+		case <-c.ctx.Done():
+			return
+		}
+	}
+}
+
+func (c *cidTask) fetchCID() error {
+	os := make([]*orm.Object, 0)
+	if err := c.db.Model(&orm.Object{}).
+		Where("cid = ?", "").
+		Find(&os).
+		Error; err != nil {
+		return err
+	}
+
+	for _, o := range os {
+		ohash, err := sha256.HashFromHex(o.Hash)
+		if err != nil {
+			return err
```

Comment:
This does not actually resolve the issue? It retries the next time and will fail again. Probably just log the skip?